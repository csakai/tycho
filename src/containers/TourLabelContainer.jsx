import React from 'react';
import TourLabel from '../components/TourLabel';

export default class TourLabelContainer extends React.Component {

  componentWillMount = () => {
    const {start, end} = this.props;

    this.setState({modifier: 'hide'});
    this.setClassAsync('show', start);
    this.setClassAsync('hide', end);
  }

  /**
   * Updates the local state class name with the given modifier at the given delay.
   *
   * @param {String} modifier - BEM CSS class modifier
   * @param {Number} timeout - delay, in milliseconds
   */
  setClassAsync = (modifier, timeout) => {
    setTimeout(() => {
      this.setState({modifier});
    }, timeout);
  }
  
  render() {
    return (
      <TourLabel
        modifier={this.state.modifier}
        text={this.props.text}
      />
    );
  }
}
