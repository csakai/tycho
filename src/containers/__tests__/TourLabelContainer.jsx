import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import TourLabelContainer from '../TourLabelContainer';

jest.useFakeTimers();

describe('App Container', () => {
  let component, tourLabelContainer;

  beforeEach(() => {
    component = shallow(<TourLabelContainer
      start={1000}
      end={3000}
      text="Hello, world"
    />);
    tourLabelContainer = component.instance();
  });

  describe('componentWillMount()', () => {
    it('should initialize the modifier in state', () => {
      expect(component.state()).toHaveProperty('modifier');
      expect(typeof component.state('modifier')).toBe('string');
    });

    it('should set the start and stop modifier change classes', () => {
      const spy = jest.spyOn(tourLabelContainer, 'setClassAsync');
      const start = 1000;
      const end = 3000;

      tourLabelContainer.props = {start, end};
      tourLabelContainer.componentWillMount();

      expect(spy).toHaveBeenCalled();
      expect(spy).toHaveBeenCalledTimes(2);
      expect(spy).toHaveBeenCalledWith('show', start);
      expect(spy).toHaveBeenCalledWith('hide', end);
    });
  });

  describe('setClassAsync()', () => {
    it('should add the given BEM modifier to the modifier after the given duration', () => {
      const modifier = 'hide';

      component.setState({modifier});
      tourLabelContainer.setClassAsync(modifier, 1000);

      jest.runAllTimers();

      expect(component.state('modifier')).toEqual(modifier);
    });
  });

  describe('render()', () => {
    it('should render the app successfully', () => {
      component.setState({time: 1});
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
