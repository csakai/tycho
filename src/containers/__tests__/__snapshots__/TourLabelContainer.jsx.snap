// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container render() should render the app successfully 1`] = `
<TourLabel
  modifier="hide"
  text="Hello, world"
/>
`;
